compiler/x/codegen/DataSnippet.cpp

```diff
@@ -54,9 +54,8 @@ TR::X86DataSnippet::addMetaDataForCodeAddress(uint8_t *cursor)
    if (_isClassAddress)
       {
       bool needRelocation = TR::Compiler->cls.classUnloadAssumptionNeedsRelocation(cg()->comp());
-      if (needRelocation)
+      if (needRelocation && !cg()->comp()->compileRelocatableCode())
```

Comment:
another commit that should have a more meaningful title and message...These changes do not "Enable checkcast inlining under AOT"

Reply:
`needRelocation` should really be renamed to `assumptionNeedsRelocation` ; I'll allow it in a subsequent PR.